pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -663,13 +663,42 @@ func (l *LibvirtDomainManager) getGuestTimeContext() context.Context {
 
 // FinalizeVirtualMachineMigration apply all post-migration changes on the domain.
 func (l *LibvirtDomainManager) FinalizeVirtualMachineMigration(vmi *v1.VirtualMachineInstance) error {
+	if err := l.hotPlugHostDevices(vmi); err != nil {
+		log.Log.Object(vmi).Reason(err).Error("failed to hot-plug host-devices")
+	}
+
 	if err := l.setGuestTime(vmi); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// hotPlugHostDevices attach host-devices to running domain
+// Currently only SRIOV host-devices are supported
+func (l *LibvirtDomainManager) hotPlugHostDevices(vmi *v1.VirtualMachineInstance) error {
+	domainName := api.VMINamespaceKeyFunc(vmi)
+	domain, err := l.virConn.LookupDomainByName(domainName)
+	if err != nil {
+		return err
+	}
+	defer domain.Free()
+
+	l.domainModifyLock.Lock()
```

Comment:
Can you please explain why `l.virConn.LookupDomainByName` doesn't need to be locked as well?

Reply:
I though that `l.domainModifyLock` should be locked when there is an operation that changes the domain
but according to the [comment at its definition](https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-launcher/virtwrap/manager.go#L115-L116) it should be locked also from "get" operations.

Reply:
It actually does not need to be locked, as it does not change anything and the information it fetches is just a descriptor to the domain. We only have one, and if something is going terribly wrong, nothing will work anyway.

But for simplicity here, I think it is fine to include it in the lock.

Reply:
I think there could be a case where during fetch domain it changes by another consumer
Why its not mandatory to lock on fetch domain operation?

Reply:
What will change? Give me an example.

Reply:
A user desided to rename the VM right after migration is finished and its time to attach host-devices

> /approve

Thank you :slightly_smiling_face: